src/main/java/com/redhat/devtools/intellij/telemetry/core/configuration/TelemetryConfiguration.java

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.telemetry.core.configuration;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.util.messages.Topic;
+import com.redhat.devtools.intellij.telemetry.core.util.Lazy;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+public class TelemetryConfiguration extends CompositeConfiguration {
+
+    public static final String KEY_MODE = "mode";
```

Comment:
The file that i stored this in is called `com.redhat.devtools.intellij.telemetry`. I'll rename the key for the mode to `com.redhat.devtools.intellij.telemetry.mode` as suggested so that it's unique among the system properties that might be passed along.